### DISCUSSION TITLE
Batch mode

### DISCUSSION THREAD

**Discussion by dondiegorivera** [cline/cline#548] [Feature Requests]
Claude supports batch mode now, what lowers input token cost significantly. It would be great to have it as an option:

https://www.anthropic.com/news/message-batches-api

**Comment by systemshift** [comment]
I was about to recommend this too. Especially when AFK, such as going for a walk or taking a break, tokens are have price. Only major change would be making cline take action without user approval, so this could be seen as something against the philosophy of the project.

**Reply by aiqbal77** [reply]
Did you hear any further on this? I've been looking for the same myself

**Comment by agabor** [comment]
I vote for this feature too. This would be so cool!

**Reply by agabor** [reply]
Reading more about the batch API, I am not convinced that Cline can use it meaningfully. Generating code for a task takes multiple messages, and each message depends on the previous result. Batch API requires you to send the messages simultaneously, of course, but at the beginning of a task, you can not know what the 3rd or 4th message will be.

**Comment by dcbartlett** [comment]
Cline (a client) talks to a service (OpenRouter or Anthropic) for it's service to work. There is no way to "Batch" a message from Cline to a service with other users' requests... Batching multiple requests together also means that a request has to wait till the batch is processed to receive a response. This is a bad user experience as the batch may not be processed for many minutes, or even hours (Cline can't control when it's processed).
